pkg/kubelet/config/common.go

```diff
@@ -55,17 +58,33 @@ func generatePodName(name string, nodeName types.NodeName) string {
 
 func applyDefaults(pod *api.Pod, source string, isFile bool, nodeName types.NodeName) error {
 	if len(pod.UID) == 0 {
-		hasher := md5.New()
-		hash.DeepHashObject(hasher, pod)
```

Comment:
In fact, according to my understanding, the hash of the pod should already be calculated in depth here, no need to get the yaml content of the pod to calculate the hash again.